Bugsnag/Breadcrumbs/BugsnagBreadcrumbs.m

```diff
@@ -137,52 +137,56 @@ - (void)setCapacity:(NSUInteger)capacity {
 }
 
 - (NSUInteger)count {
-    return self.breadcrumbs.count;
+    __block NSUInteger count;
+    dispatch_barrier_sync(self.readWriteQueue, ^{
+        count = self.breadcrumbs.count;
+    });
+    return count;
 }
 
 - (BugsnagBreadcrumb *)objectAtIndexedSubscript:(NSUInteger)index {
-    if (index < [self count]) {
-        __block BugsnagBreadcrumb *crumb = nil;
-        dispatch_barrier_sync(self.readWriteQueue, ^{
-          crumb = self.breadcrumbs[index];
-        });
-        return crumb;
-    }
-    return nil;
+    __block BugsnagBreadcrumb *crumb = nil;
+    dispatch_barrier_sync(self.readWriteQueue, ^{
+        if (index < self.breadcrumbs.count) {
+            crumb = self.breadcrumbs[index];
+        }
+    });
+    return crumb;
 }
 
 - (NSArray *)arrayValue {
-    if ([self count] == 0) {
-        return nil;
-    }
-    __block NSMutableArray *contents =
-        [[NSMutableArray alloc] initWithCapacity:[self count]];
+    __block NSMutableArray *contents = nil;
     dispatch_barrier_sync(self.readWriteQueue, ^{
```

Comment:
I'm not convinced that a dispatch barrier is necessary - being defensive and serializing a `mutableCopy` of `self.breadcrumbs` should be sufficient IMO.